### ISSUE TITLE
Ensure sufficient testing of controllers

### ISSUE THREAD

**Issue by danryan** [revily/revily#16]
Investigate current testing of controllers and add more examples where necessary.

**Comment by pfhawkins** [comment]
Is this an issue any more? controller specs seem to be running quite speedily and are certainly not the biggest contributor to the 2-minute spec runtime:

<pre>{8:48}[2.0.0-p247]~/projects/reveille:master ✗ ➭ b rspec spec/controllers
No DRb server is running. Running in local process instead ...
You are using WebMock 1.13.0. VCR is known to work with WebMock >= 1.8.0, < 1.12. It may not work with this > version.
Run options:
 include {:focus=>true}
 exclude {:external=>true}
All examples were filtered out; ignoring {:focus=>true}
*.
 
Pending:
  V1::UsersController
    # No reason given
    # ./spec/controllers/users_controller_spec.rb:5
Finished in 0.27462 seconds
2 examples, 0 failures, 1 pending</pre>

I could, of course, be barking up the wrong tree.

**Comment by pfhawkins** [comment]
<pre>b rspec</pre> is just an alias to <pre>bundle exec rspec</pre>.

I am running these on a 2013 macbook air.

**Comment by danryan** [comment]
I don't think so anymore. I've sort of ignored controller specs, opting instead to focus on flexing the API itself (controllers simply being the endpoint defining the API, after all). Those are in spec/api and test the stack by issuing real HTTP requests with JSON payloads. 

Most of the specs are actually rather speedy -- less than a second for the slowest examples -- with the exception of `user_schedule_spec.rb`. That beast takes a blistering 30 seconds to run on average:

```
Top 10 slowest examples (40.18 seconds, 46.7% of total time):
  UserSchedule#initialize builds a user schedule
    35.57 seconds ./spec/models/user_schedule_spec.rb:19
  Schedule#current_user_on_call weekly rotation with lots of users works with a large number of users
    0.86773 seconds ./spec/models/schedule_spec.rb:66
  Incident states acknowledge can transition to :triggered
    0.51486 seconds ./spec/models/incident_spec.rb:120
  Incident states escalate can transition from :triggered
    0.48458 seconds ./spec/models/incident_spec.rb:82
  Incident states initial state should be pending
    0.47879 seconds ./spec/models/incident_spec.rb:44
  Incident states acknowledge cannot transition to :acknowledged
    0.47521 seconds ./spec/models/incident_spec.rb:114
  Incident states escalate can transition from :acknowledged
    0.45799 seconds ./spec/models/incident_spec.rb:90
  Schedule#current_user_on_call daily rotation works with multiple users
    0.4498 seconds ./spec/models/schedule_spec.rb:35
  services PUT /services/:id/enable should respond with :no_content
    0.44845 seconds ./spec/api/v1/services_spec.rb:52
  Incident states acknowledge can transition to :resolved
    0.42699 seconds ./spec/models/incident_spec.rb:127
```

I haven't ripped into that test yet to see what's making it slow. It could be that the code itself is slow. `UserSchedule` handles determining which user is currently on-call for a given schedule. Ruby isn't known for fast date parsing, but it's more likely that I'm not doing this in the most efficient manner :grinning:

**Comment by danryan** [comment]
I've been considering re-adding controller specs as I have introduced some more complexity into them lately. Refactoring this issue into "ensure sufficient testing of controllers".

**Comment by danryan** [comment]
Opened #30 to deal with the UserSchedule slowness.
